.gitlab-ci.yml

```diff
@@ -0,0 +1,55 @@
+image: python:3.6
+
+stages:
+  - infra
+  - apps
+  - test
+  - deploy
+
+before_script:
+  - echo "Running Before Script"
+  - source environment
+  - apt-get -y update
+  - apt-get -y install jq moreutils curl unzip zip
+  - mkdir ./external_binaries
+  - curl https://releases.hashicorp.com/terraform/0.11.7/terraform_0.11.7_linux_amd64.zip -o /tmp/terraform.zip
+  - unzip /tmp/terraform.zip -d ./external_binaries/
+  - rm /tmp/terraform.zip
+  - export PATH=$PATH:`pwd`/external_binaries/
+  - pip install awscli --upgrade
+  - pip install boto3 --upgrade
+  - pip install virtualenv
+  - make secrets
+  - export GOOGLE_APPLICATION_CREDENTIALS="${PROJECT_ROOT}/gcp-credentials-logs-travis.json"
+
+infra:
+  stage: infra
+  script:
+    - echo "Infra-plan stage"
```

Comment:
nit: I would remove the `echo`s. In gitlab these are broken up by stage anyway. Might as well keep it super clean.